src/app/backend-api.service.ts

```diff
@@ -31,6 +44,32 @@ export class BackendAPIService {
       },
     );
   }
+
+  GetUserProfiles(
+    publicKeys: any[]
+  ): Observable<{[key: string]: UserProfile}> {
+    return new Observable<{[key: string]: UserProfile}>(subscriber => {
+      const userProfiles: {[key: string]: any} = {};
+
+      if (publicKeys.length > 0) {
+        for (const publicKey of publicKeys) {
+          userProfiles[publicKey] = {};
+        }
+        this.GetUsersStateless(publicKeys).subscribe(res => {
+          for (const user of res.UserList) {
+            userProfiles[user.PublicKeyBase58Check] = {
+              username: user.ProfileEntryResponse?.Username,
+              profilePic: user.ProfileEntryResponse?.ProfilePic,
+            };
+          }
+          subscriber.next(userProfiles);
```

Comment:
While this implementation is fine, it's not ideal. If you want to try your hand at RXJS operators (which can be annoying but very powerful), checkout `backend-api.ts` in the frontend repo. The best way to implement this type of behavior is to do something akin to:
```
const userProfiles: {[key: string]: any} = {};
const req = this.GetUsersStateless(publicKeys);
if () {
  req.pipe(
    map((res) => {
      // ...
      return userProfiles;
    })
  )
} else {
  return of(userProfiles);
}

```

Reply:
ugh, RXJS... great comment though, thanks